[BB-1637] Pass the referer from the client to the YouTube API

lgp171188 opened the pull request:
The changes introduced in https://github.com/edx/edx-platform/commit/3136b3ecd3bc0d375e8e3f47f850805dc19b70db moved the YouTube Data API call which was being made in the JavaScript code to a proxy API. Whenever the API calls were made from the client-side, the referer header was automatically set by the browser and with the move to a server-side API call, this is no longer the case. This causes the API calls using API keys requiring a referer (and often whitelisted to be allowed only from specific domains) to fail. Due to this, the courseware pages containing YouTube videos to show an alert message stating

    You have been logged out of your edX account. Click Okay to log in again now. Click Cancel to stay on this page (you must log in again to save your work).

In such scenarios, a warning message is logged in the browser console and also in the LMS logs.

Below is an example message from the LMS logs

```
[views.py:301] - YouTube API request failed with status code=403 - Error message is={
 "error": {
  "errors": [
   {
    "domain": "usageLimits",
    "reason": "ipRefererBlocked",
    "message": "The request did not specify any referer. Please ensure that the client is sending referer or use the API Console to remove the referer restrictions.",
    "extendedHelp": "https://console.developers.google.com/apis/credentials?project=XXXXXXX"
   }
  ],
  "code": 403,
  "message": "The request did not specify any referer. Please ensure that the client is sending referer or use the API Console to remove the referer restrictions."

```

This PR fixes this issue by passing the referer header from the client, when available, to the server-side API call.

**JIRA tickets**: OSPR-3813

**Dependencies**: None

**Sandbox URL**: TBD - sandbox is being provisioned.

**Merge deadline**: None 

**Testing instructions**:

* Visit the Google Developers Console at https://developers.google.com. Log in to an existing account or create a new Google Developer account.
* Create a new project in the Google Developers Console:
https://console.developers.google.com/
* Go to the APIs overview section and click the + Enable API
button. 
* Select YouTube Data API, then click the > Enable button.
* Click the Create Credentials button to add API credentials to your
project. Look for the following text: "If you wish you can skip this
step and create an API key, client ID, or service account" and click
the API key link.
* Enter a recognizable name, and select *HTTP referrers (web sites)
under Key restriction. Enter appropriate wildcard pattern to only
and allow requests from the hostname for your edxapp instance.  E.g.,
if the instance URL is www.abc.com, allow requests from:
*.abc.com/*
* Copy the generate access key and set it to the
EDXAPP_YOUTUBE_API_KEY variable and deploy the instance without this fix.
* Visit a courseware page with a YouTube video. The error condition mentioned above in the description occurs.
* Apply the fix in this PR, restart the LMS.
* Visit the same courseware page with a YouTube video.
* Verify that it works okay and there are no error conditions or warning messages any more.

**Reviewers**
- [ ] @viadanna 
- [ ] edX reviewer[s] TBD

edx-webhook commented:
Thanks for the pull request, @lgp171188! I've created [OSPR-3813](https://openedx.atlassian.net/browse/OSPR-3813) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

natabene commented:
@lgp171188 Thank you for your contribution, please let me know once it is ready for our review.

lgp171188 commented:
@natabene, this is ready for edX review.

natabene commented:
@ormsbee This is ready for your review.

lgp171188 commented:
@ormsbee, did you get a chance to check this out?

lgp171188 commented:
@ormsbee, did you get a chance to check this out?

lgp171188 commented:
@ormsbee, did you get a chance to check this out?

natabene commented:
@lgp171188 Sorry we don't have bandwidth to review right now. Please reach out to @pomegranited for prioritization.

giovannicimolin commented:
jenkins run all

ormsbee commented:
jenkins run all

lgp171188 commented:
@giovannicimolin, (CC @viadanna), @ormsbee, I have rebased the PR on the master branch and fixed/tweaked the original changes to ensure that they work after the rebase. The CI checks have passed. This is ready for edX review and can be picked up as per the prioritization.

edx-webhook commented:
@lgp171188 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/edx-platform&entry.867055334=https://github.com/edx/edx-platform/pull/21507&entry.1484655318=https://github.com/lgp171188&entry.752974735=2019-08-29+18:29&entry.1917517419=2020-05-11+15:46&entry.2133058324=Yes) so we can improve your experience in the future.

lgp171188 commented:
@nedbat, thanks for reviewing and merging this PR :)
